### PR TITLE
EDPLAT-576 subscribe to lambda to galileo topic

### PR DESCRIPTION
**Pull request by kodjobaah**

### CODE REVIEW

File: infrastructure/src/babel/buildenvironment.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
+# -*- coding: utf-8 -*-
```
Comment (timsweetman) [inline infrastructure/src/babel/buildenvironment.py]: This looks like a duplicate of environmentbuilder.py. If it's not in use, please delete.

**Comment by timsweetman** [comment]
I run babel.environmentbuilder, and get this error:
* It thinks the permission is called "galilioBabelIntint". Too many "int"s
* Presumably it should notice that there's already a statement called that, and continue, or handle the error somehow

    python3 -m babel.environmentbuilder --lambda-function-bucket galileo-babel-lambda --environment int --aws-accountId 195048873603 --region eu-west-2 --galileo-accountId 161201357662 --galileo-region eu-west-1 --galileo-topic int-galileo-babel-resources-PublishTopic-ASEPFH7ABVQ2
    Nothing to update
    ---source-arc[eu-west-1:161201357662:int-galileo-babel-resources-PublishTopic-ASEPFH7ABVQ2
    Traceback (most recent call last):
      File "/usr/lib64/python3.4/runpy.py", line 170, in _run_module_as_main
        "__main__", mod_spec)
      File "/usr/lib64/python3.4/runpy.py", line 85, in _run_code
        exec(code, run_globals)
      File "/mnt/hgfs/workspace/editorial-search-galileo-babel/infrastructure/src/babel/environmentbuilder.py", line 144, in <module>
        main()
      File "/mnt/hgfs/workspace/editorial-search-galileo-babel/infrastructure/src/babel/environmentbuilder.py", line 140, in main
        galileoPermisionAndSubscription.add_permisions(environment, region, awsAccountId)
      File "/mnt/hgfs/workspace/editorial-search-galileo-babel/infrastructure/src/babel/environmentbuilder.py", line 71, in add_permisions
        Qualifier=str(env))
      File "/home/vagrant/galileo-babel/lib/python3.4/site-packages/botocore/client.py", line 314, in _api_call
        return self._make_api_call(operation_name, kwargs)
      File "/home/vagrant/galileo-babel/lib/python3.4/site-packages/botocore/client.py", line 612, in _make_api_call
        raise error_class(parsed_response, operation_name)
    botocore.errorfactory.ResourceConflictException: An error occurred (ResourceConflictException) when calling the AddPermission operation: The statement id (galilioBabelIntint) provided already exists. Please provide a new statement id, or remove the existing statement.

File: infrastructure/src/babel/galileobabelstack.py [hunk 2]
```diff
@@ -20,6 +20,17 @@ def addBucket(self, template):
...
+    def add_permisions(self, template, aws_lambda, galileoAccountId, galileoTopic, galileoRegion):
+            galileoBabelTopic = "arn:aws:sns:"+galileoRegion+":"+galileoAccountId+":"+galileoTopic
+            template.add_resource(Permission(
+                "InvokeLambdaPermission",
+                FunctionName=GetAtt(aws_lambda, "Arn"),
+                Action="lambda:InvokeFunction",
+                SourceArn = galileoBabelTopic,
+                Principal= "sns.amazonaws.com"
+            ))
+
```
Comment (timsweetman) [inline infrastructure/src/babel/galileobabelstack.py]: Presumably unused now - please remove.

File: infrastructure/src/babel/environmentbuilder.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+        print("Bucket does not exist -- addding")
+        galileoBabelStack.addBucket(t)
+    
+
+    galileoTopic = params[6]
+    galileoAccountId = params[4]
+    galileoRegion = params[5]
+    
+    aws_lambda = galileoBabelStack.build(t)
+    #galileoBabelStack.add_permisions(t, aws_lambda, galileoAccountId,galileoTopic, galileoRegion)
```
Comment (timsweetman) [inline infrastructure/src/babel/environmentbuilder.py]: Please remove commented-out code.

**Comment by kodjobaah** [comment]
@timsweetman 
Can you try this instead..
```
python3 -m babel.environmentbuilder --lambda-function-bucket galileo-babel-lambda --environment mathew --aws-accountId 195048873603 --region eu-west-2 --galileo-accountId 161201357662 --galileo-region eu-west-1 --galileo-topic int-galileo-babel-resources-PublishTopic-ASEPFH7ABVQ2
```

**Comment by timsweetman** [comment]
That runs without an error, because the "mathew" environment is new.

**Comment by kodjobaah** [comment]
@timsweetman i have address the issue when the stack already exists and you try to add permissions.. 
thanx..

File: README.md [hunk 4]
```diff
@@ -36,7 +36,11 @@ You will need to install troposhere and some other Python packages:
     
     In the directory  ```<APP_DIRECTORY>/infrastructure/src``` 
     
-    Type the following:```python -m babel.devenvironment --lambda-function-bucket <s3-bucket-containing-lambda-function-zipped> --environment <env> --aws-accountId <aws-account-id> --region <aws-region>```
+    Type the following:
+    ```python -m babel.environmentbuilder --lambda-function-bucket <s3-bucket-containing-zipped-lambda-function> --environment <env> --aws-accountId <aws account id> --region <aws region> --galileo-accountId <galileo accountId> --galileo-region <galileo region> --galileo-topic <galileo topic>```
+
```
Comment (timsweetman) [inline README.md]: This is too much configuration to deal with on the command line, where all of these are basically constants per-environment (unless the Galileo maintainers change them).

I think the user should specify *env* on the command line (int, test, live, or a throwaway environment like "test2" or "tims"), and the code should then pick the galileo account number (one of two), galileo region (always eu-west-1 for the foreseeable future), and galileo topic. If those values are documented on Confluence or somewhere, there should be a link to whatever that page is.

So for the user, the command would be something like

     python -m babel.environmentbuilder int
     python -m babel.environmentbuilder test
     python -m babel.environmentbuilder throwawaytest4

**Comment by timsweetman** [comment]
To repeat: Please remove commented-out code.

**Comment by timsweetman** [comment]
Lastly, spelling/typos:

* gallileo should be galileo
* permision should be permission
* permssion should be permission

File: README.md (hunk 4 above)
Comment (kodjobaah) [reply README.md]: @timsweetman ` python -m babel.environmentbuilder throwawaytest4` so in this particular case what is the environment.

Comment (timsweetman) [reply README.md]: The environment is "throwawaytest4", and I'd expect this to use *int*'s settings for the Galileo topic and account.

Comment (kodjobaah) [reply README.md]: @timsweetman  Have made the changes..
The lamba function bucket is the only thing that appears not to be a constant..based on your assumptions..

```python3 -m babel.environmentbuilder --env throwawaytest4 --lambda-function-bucket galileo-babel-lambda```

Comment (kodjobaah) [reply README.md]: @timsweetman ~~~just realised that still need to change the readme.md..~~~

**Comment by timsweetman** [comment]
👍 Runs OK when pointed at int. Haven't tried other environments. Code looks good. Will merge when you're ready.

**Comment by kodjobaah** [comment]
@timsweetman just removed some unused imports and code. So I think it is ready to be merged.. thnx.

**Pull request merged**